src/selectors/per-thread/thread.js

```diff
@@ -45,6 +46,11 @@ export function getThreadSelectorsPerThread(threadIndex: ThreadIndex): * {
     getThread(state).stringTable;
   const getSamplesTable: Selector<SamplesTable> = state =>
     getThread(state).samples;
+  const getThreadRange: Selector<StartEndRange> = createSelector(
+    getThread,
+    ProfileSelectors.getProfileInterval,
+    ProfileData.getTimeRangeForThread
+  );
```

Comment:
I'm not super happy about this, because we'll compute `getTimeRangeForThread` twice for each thread: once here, and once when running `getTimeRangeIncludingAllThreads` when loading the profile, to compute the `rootRange` reducer.

I considered various things to work around this; I think the most sensible would be to use a memoize operation on `getTimeRangeForThread`. But I wanted to wait for your feedback before doing it, as the performance didn't seem so bad to me actually. It's also possible to do it later if we find this is a problem.

Reply:
I don't really have the problem loaded in my head well enough, but ideally things would be memoized by selectors only. However, I'm guessing there's some dependency issue here, so a memoized function would be the next step.

I'd be fine with a memoized function and a comment explaining why.

Reply:
Currently the `rootRange` is computed by a reducer, when the action `PROIFLE_LOADED` happens. It was surprising to me at first, as I expected this to be computed in a selector instead. It could be computed by the new per-thread selector of `getThreadRange`.
But then I thought we could enter in a dependency hell, as some other per-thread selectors depend themselves on the `rootRange`... and I didn't want to change the world now -- even if I knew how to change this in a proper way.